language/move-prover/tests/sources/restrictions.move

```diff
@@ -0,0 +1,60 @@
+// no-boogie-test
+module M {
+
+    struct S {
+        f: u64
+    }
+
+    spec module {
+        // Tuples as result type.
+        define f1(): (num, num) { (1, 2) }
+
+        // Functions as result type.
+        define f2(): | |num { | | 1 }
```

Comment:
I think I understand the syntax for the definition of f1 above.  But f2 is blowing my head.

Reply:
There is a small bug with empty parameter lists in the parser. When this is fixed it is easier.